projects/ngx-avatar/src/lib/avatar.component.ts

```diff
@@ -133,8 +133,8 @@ export class AvatarComponent implements OnChanges, OnDestroy {
     for (const propName in changes) {
       if (this.avatarService.isSource(propName)) {
         const sourceType = AvatarSource[propName.toUpperCase()];
-        if (changes[propName].currentValue) {
-          const currentValue = changes[propName].currentValue;
+        const currentValue = changes[propName].currentValue;
+        if (currentValue && typeof currentValue === 'string') {
```

Comment:
No need to check both, just `typeof` alone will do:

```suggestion
        if (typeof currentValue === 'string') {
```

Reply:
Actually we want to check for empty string too, so we need a second check. But we could make it more explicit if you prefer

Reply:
Oh I see, I thought you were just checking if the variable had a valid string value because I'm so used to test `typeof` first and then check the content of the variable.